Public interfaces FullTextSearch

ArtificialOwl opened the pull request:
Signed-off-by: Maxence Lange <maxence@artificial-owl.com>

ChristophWurst commented on lib/private/FullTextSearch/FullTextSearchManager.php:
```diff
@@ -0,0 +1,221 @@
+
+	/**
+	 * @param string $providerId
+	 * @param string $documentId
```
is that a path?

daita replied:
no its an Id, could be a fileId, a bookmarkId, any Id

ChristophWurst commented on lib/private/FullTextSearch/FullTextSearchManager.php:
```diff
@@ -0,0 +1,221 @@
+	/**
+	 * @param string $providerId
+	 * @param string $documentId
+	 * @param string $ownerId
```
is that a UID? If so, I'd suggest to rename

daita replied:
I could rename it to userId

daita replied:
renamed

ChristophWurst commented on lib/private/FullTextSearch/FullTextSearchManager.php:
```diff
@@ -0,0 +1,221 @@
+	 * @param string $providerId
+	 * @param string $documentId
+	 * @param string $ownerId
+	 * @param int $status
```
are the allowed values documented anywhere? could you add these as interface constants?

daita replied:
added a link to the interface

ChristophWurst commented on lib/public/FullTextSearch/Model/IIndex.php:
```diff
@@ -0,0 +1,292 @@
+	public function setOwnerId(string $ownerId): IIndex;
+
+	/**
+	 * Get the owner of the originl document.
```
typo

ChristophWurst reviewed (commented):
Structure looks good :heavy_check_mark: 
Private API is minimal :heavy_check_mark: 
Public classes are final :heavy_check_mark: 

Can't review the semantics of the APIs because I lack the full text search knowledge.

Found a few remarks in the code: naming and a typo.

rullzer commented on lib/public/FullTextSearch/IFullTextSearchManager.php:
```diff
@@ -0,0 +1,190 @@
+	 * @param int $status
+	 * @param bool $reset
+	 *
+	 * @return mixed
```
Is this true?

daita replied:
wow, thanks. does not returns anything.

rullzer commented on lib/public/FullTextSearch/IFullTextSearchManager.php:
```diff
@@ -0,0 +1,190 @@
+	 * @param int $status
+	 * @param bool $reset
+	 *
+	 * @return mixed
```
Same here

daita replied:
wow, thanks. does not returns anything.

rullzer reviewed (approved):
Looks good to me. :rocket:

ArtificialOwl committed:
public interfaces FullTextSearch

Signed-off-by: Maxence Lange <maxence@artificial-owl.com>
-@return mixed

Signed-off-by: Maxence Lange <maxence@artificial-owl.com>
Merge remote-tracking branch 'origin/interfaces-fulltextsearch-2' into interfaces-fulltextsearch-2

Signed-off-by: Maxence Lange <maxence@artificial-owl.com>
